pymodaq_plugins/daq_viewer_plugins/plugins_1D/daq_1Dviewer_LecroyWaverunner6Zi.py

```diff
@@ -1,202 +1,223 @@
+        To prevent from this error the user should stop the pymodaq acquisition (STOP button in the GUI interface),
+        then change the oscilloscope parameter of his choice, then rerun the acquisition. See also the comments in the
+        grab_data function below.
+
+Issues
+------
+
+If you see any misbehavior you can raise an issue on the github repository :
+
+    https://github.com/CEMES-CNRS/pymodaq_plugins/issues
+
+"""
+
+class DAQ_1DViewer_LecroyWaverunner6Zi(DAQ_Viewer_base):
+    """
+        ==================== ========================
+        **Attributes**        **Type**
+        *data_grabed_signal*  instance of pyqtSignal
+        *VISA_rm*             ResourceManager
+        *com_ports*
+        *params*              dictionnary list
+        *keithley*
+        *settings*
+        ==================== ========================
+    """
+    data_grabed_signal = pyqtSignal(list)
+
+    ##checking VISA ressources
+    VISA_rm = ResourceManager()
+    resources = list(VISA_rm.list_resources())
+
+    params = comon_parameters + [
+            {'title': 'VISA:',
+             'name': 'VISA_ressources',
+             'type': 'list',
+             'values': resources},
+            {'title': 'Channels:',
+             'name': 'channels',
+             'type': 'itemselect',
+             'value': dict(all_items=["C1", "C2", "C3", "C4", "F1", "F2", "F3", "F4"], selected=["C1"])},
+        ]
+
+    def __init__(self, parent = None, params_state = None):
+        super(DAQ_1DViewer_LecroyWaverunner6Zi, self).__init__(parent, params_state)
+        self.controller = None
+
+    def ini_detector(self, controller = None):
+        """
+            Initialisation procedure of the detector.
+
+            Returns
+            -------
+
+                The initialized status.
+
+            See Also
+            --------
+            daq_utils.ThreadCommand
+        """
+        self.status.update(edict(initialized = False, info = "", x_axis = None, y_axis = None, controller = None))
+        try:
+            if self.settings.child(('controller_status')).value() == "Slave":
+                if controller is None:
+                    raise Exception('no controller has been defined externally while this detector is a slave one')
+                else:
+                    self.controller = controller
+            else:
+                self.controller = active_dso
+                usb_address = "USBTMC:" + self.settings.child(('VISA_ressources')).value()
+                self.controller.MakeConnection(usb_address)
+                # set the timeout of the scope to 10 seconds
+                # may be not needed
+                self.controller.setTimeout(10)
+
+            self.status.initialized = True
+            self.status.controller = self.controller
+            # self.controller.WriteString(r"""vbs 'app.acquisition.triggermode = "stopped" ' """, 1)

```

Comment:
This has been added for tests I suppose

Reply:
what should I do then, merge it into your plugin?

Reply:
wait for the author of the commit to make the corrections before merging